pypgstac/pypgstac/load.py

```diff
@@ -1,246 +1,501 @@
+                            """
+                        ).format(sql.Identifier(partition["partition"]))
+                    )
+                    if insert_mode in ("ignore", "insert_ignore"):
+                        cur.execute(
+                            sql.SQL(
+                                """
+                                INSERT INTO {}
+                                SELECT *
+                                FROM items_ingest_temp ON CONFLICT DO NOTHING;
+                                """
+                            ).format(sql.Identifier(partition["partition"]))
+                        )
+                    elif insert_mode == "upsert":
+                        cur.execute(
+                            sql.SQL(
+                                """
+                                INSERT INTO {} SELECT * FROM items_ingest_temp
+                                ON CONFLICT (id) DO UPDATE
+                                SET
+                                    datetime = EXCLUDED.datetime,
+                                    end_datetime = EXCLUDED.end_datetime,
+                                    geometry = EXCLUDED.geometry,
+                                    collection = EXCLUDED.collection,
+                                    content = EXCLUDED.content
+                                ;
+                            """
+                            ).format(sql.Identifier(partition["partition"]))
+                        )
+                    elif insert_mode == "delsert":
+                        cur.execute(
+                            sql.SQL(
+                                """
+                                WITH deletes AS (
+                                    DELETE FROM items i USING items_ingest_temp s
+                                        WHERE
+                                            i.id = s.id
+                                            AND i.collection = s.collection
+                                            AND i IS DISTINCT FROM s
+                                    RETURNING i.id, i.collection
+                                )
+                                INSERT INTO {}
+                                SELECT s.* FROM
+                                    items_ingest_temp s
+                                    JOIN deletes d
+                                    USING (id, collection);
+                                ;
+                            """
+                            ).format(sql.Identifier(partition["partition"]))
+                        )
+                else:
+                    raise Exception(
+                        "Available modes are insert, ignore, upsert, and delsert."
+                        f"You entered {insert_mode}."
+                    )
+        logging.debug(
+            f"Copying data for {partition} took {time.perf_counter() - t} seconds"
+        )
 
+    def load_items(
+        self,
+        file: Optional[str] = "stdin",
```

Comment:
This loses the ability to pass in an `Iterator[Dict[str, Any]]` or `Iterator[bytes]` which we were using. Can we make this loader similarly work with things other than files?